packages/teams-js/src/internal/communication.ts

```diff
@@ -320,6 +324,45 @@ export function sendMessageToParentAsync<T>(actionName: string, args: any[] | un
   });
 }
 
+/**
+ * @hidden
+ * Send a message to parent requesting a MessageChannel Port.
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function requestPortFromParentWithVersion(
+  apiVersionTag: string,
+  actionName: string,
+  args: any[] | undefined = undefined,
+): Promise<MessagePort> {
+  if (!isFollowingApiVersionTagFormat(apiVersionTag)) {
+    throw Error(
+      `apiVersionTag: ${apiVersionTag} passed in doesn't follow the pattern starting with 'v' followed by digits, then underscore with words, please check.`,
+    );
+  }
+  return new Promise((resolve) => {
+    const request = sendMessageToParentHelper(apiVersionTag, actionName, args);
+    resolve(waitForPort(request.id));
+  });
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function waitForPort(requestId: number): Promise<MessagePort> {
+  return new Promise<MessagePort>((resolve, reject) => {
+    CommunicationPrivate.portCallbacks[requestId] = (port: MessagePort | undefined, args?: any[]) => {
```

Comment:
>undefined [](http://example.com/codeflow?start=73&length=9)

This callback accepts a type of `MessagePort | undefined` as its first parameter, but in the usage on line 765, it looks like the possible values being passed are `MessagePort | null`

(I think this could be fixed in a spirit similar to my earlier comment about the type used when declaring `portCallbacks` ?)

Reply:
Yep, this is proof that loosely typing the functions is a bad idea. Updated the typing and this instance to be undefined in all places.